src/traces/funnel/style.js

```diff
@@ -31,6 +31,7 @@ function style(gd, cd, sel) {
             }
         });
 
+        Drawing.pointStyle(gTrace.selectAll('path'), trace, gd);
```

Comment:
Does this call make the loop above, or some of it, moot?

Reply:
The `trace` variable uses `each` loop.
I think this part is similar to what `bar` does [here](https://github.com/plotly/plotly.js/blob/3f338292575425242bdd6f0e0b3548c80d039028/src/traces/bar/style.js#L48).

Reply:
It just looked to me as though at least the `Color.fill` and `Color.stroke` calls (L27-28) might be duplicated by `pointStyle`. Not `Drawing.dashLine` and I can't really tell about `opacity`... but the point is `Drawing.pointStyle` does more than apply a pattern, yet the other things it does already worked for funnels prior to this PR.

Reply:
Actually deselect style is broken on `funnel` now.
Also strangely if you make a selection on `barpolar` traces of `pattern_bars` mock, the points from `bar` traces would be selected!

Reply:
@s417-lama I am about to log off for today. Would you mind investigating this part?

Reply:
I investigated the selection issue a little bit.
Seems like determining candidates for selection (`determineSearchTraces` function in `src/plots/cartesian/select.js`) is broken when polar plot and other plots coexist.

When we try to select `barpolar` plots, `bar` plots are wrongly included in the candidate lists for selection.
In `determineSearchTraces` fn, the following line is executed for `bar` plots, which is an inappropriate behavior.

https://github.com/plotly/plotly.js/blob/3f338292575425242bdd6f0e0b3548c80d039028/src/plots/cartesian/select.js#L663

This bug happens because the values of `xAxisIds` and `yAxisIds` are set to `["x"]` and `["y"]` in `barpolar` plots, but by nature polar plots should not have cartesian axes. As a result, the plot which has `x` and `y` axes, that is `bar` plot in this case, is incorrectly included in `searchTraces`. I suggest to clear x-axes and y-axes in case of `barpolar` plots somewhere in codes.

Reply:
`pattern` removed from `funnel` attributes in https://github.com/plotly/plotly.js/pull/5537/commits/4c4828a60ad8f2456d83223b74013cb303bf5b0b. So there is no need to investigate this in the PR.